### PR TITLE
polkadot companion #12608 (Pools claim permissions)

### PR DESCRIPTION
**Pull request by rossbulat**
Updates weights for the added pool claims functions and logic.

- `bond_extra_reward` has been renamed to `bond_extra_other`, and accounts for `ClaimPermission` check.
- `claim_payout` now accounts for `ClaimPermission` check.
- Added `set_claim_permission` weights.

### CODE REVIEW

**Comment by rossbulat** [comment]
/cmd queue -v PATCH_substrate=12608 -c bench-bot $ runtime runtime pallet_nomination_pools

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2421565

**Comment by rossbulat** [comment]
bot bench -v PATCH_substrate= 12608 $ runtime kusama-dev pallet_nomination_pools

**Comment by rossbulat** [comment]
bot cancel 12-0404327b-7d53-4d99-9495-f1239b9d9de9

**Comment by rossbulat** [comment]
bot clean

**Comment by rossbulat** [comment]
bot help

**Comment by rossbulat** [comment]
bot bench -v PATCH_substrate=12608 $ runtime polkadot-dev pallet_nomination_pools

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2422570

**Comment by rossbulat** [comment]
bot bench -v PATCH_substrate=12608 $ runtime kusama-dev pallet_nomination_pools

**Comment by rossbulat** [comment]
bot rebase

**Comment by rossbulat** [comment]
bot bench -v PATCH_substrate=12608 $ runtime kusama-dev pallet_nomination_pools

**Comment by rossbulat** [comment]
bot bench -v PATCH_substrate=12608 $ runtime westend-dev pallet_nomination_pools

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2423441

**Comment by rossbulat** [comment]
bot rebase

**Comment by rossbulat** [comment]
bot clean

**Comment by rossbulat** [comment]
bot bench -v PATCH_substrate=12608 $ runtime kusama-dev pallet_nomination_pools

**Comment by rossbulat** [comment]
bot rebase

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2426694

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2426883

File: runtime/polkadot/src/weights/pallet_nomination_pools.rs [hunk 1]
```diff
@@ -16,21 +16,24 @@
 //! Autogenerated weights for `pallet_nomination_pools`
 //!
 //! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 4.0.0-dev
-//! DATE: 2023-01-23, STEPS: `50`, REPEAT: 20, LOW RANGE: `[]`, HIGH RANGE: `[]`
-//! HOSTNAME: `runner-b3zmxxc-project-163-concurrent-0`, CPU: `Intel(R) Xeon(R) CPU @ 2.60GHz`
+//! DATE: 2023-02-21, STEPS: `50`, REPEAT: `20`, LOW RANGE: `[]`, HIGH RANGE: `[]`
+//! WORST CASE MAP SIZE: `1000000`
+//! HOSTNAME: `runner-ehxwxxsd-project-163-concurrent-0`, CPU: `Intel(R) Xeon(R) CPU @ 2.60GHz`
 //! EXECUTION: Some(Wasm), WASM-EXECUTION: Compiled, CHAIN: Some("polkadot-dev"), DB CACHE: 1024
```
Comment (gpestana) [inline runtime/polkadot/src/weights/pallet_nomination_pools.rs:22]: it seems that the old weights were generated outside the bot machine 🤔

**Review by gpestana — APPROVED** [review]
LGTM

**Review by Ank4n — APPROVED** [review]

File: runtime/polkadot/src/weights/pallet_nomination_pools.rs (hunk 1 above)
Comment (rossbulat) [reply runtime/polkadot/src/weights/pallet_nomination_pools.rs:22]: Interesting, they did look outdated, much less commenting and no mentioning of proofs. I imagine it was a much lower version of what is available now.

**Review by ruseinov — APPROVED** [review]
the weights look legit!

**Comment by rossbulat** [comment]
bot fmt

File: runtime/polkadot/src/weights/pallet_nomination_pools.rs (hunk 1 above)
Comment (rossbulat) [reply runtime/polkadot/src/weights/pallet_nomination_pools.rs:22]: Cool, I guess they'll be replaced upon the runtime upgrade with the latest version also.

**Comment by rossbulat** [comment]
bot rebase

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2428041

File: .cargo/config.toml [hunk 2]
```diff
@@ -29,4 +29,4 @@ rustflags = [
   "-Aclippy::needless_option_as_deref",  # false positives
   "-Aclippy::derivable_impls",           # false positives
   "-Aclippy::stable_sort_primitive",     # prefer stable sort
-]
+]
```
Comment (kianenigma) [inline .cargo/config.toml:32]:
```suggestion
]

```

**Review by kianenigma — APPROVED** [review]

**Pull request merged**